aisuite/providers/sambanova_provider.py

```diff
@@ -0,0 +1,32 @@
+import os
+from aisuite.provider import Provider
+from openai import OpenAI
+
+
+class SambanovaProvider(Provider):
+    def __init__(self, **config):
+        """
+        Initialize the SambaNova provider with the given configuration.
+        Pass the entire configuration dictionary to the OpenAI client constructor.
+        """
+        # Ensure API key is provided either in config or via environment variable
+        config.setdefault("api_key", os.getenv("SAMBANOVA_API_KEY"))
+        if not config["api_key"]:
+            raise ValueError(
+                "Sambanova API key is missing. Please provide it in the config or set the SAMBANOVA_API_KEY environment variable."
+            )
+
+        # Pass the entire config to the OpenAI client constructor
```

Comment:
You wrote in the comment, but I don't think you are passing the entire config.
Assuming that is what you want, you could add/replace the base_url in config and pass the entire thing to OpenAI client.

Reply:
@snova-zoltanc  - please address above. We can merge it after that. Thank you.

Reply:
@rohitprasad15 Thanks for the review I made the update